### PR TITLE
fix issues where funcs are defined out of order

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: symbol_exporter/ast_symbol_extractor.py [hunk 1]
```diff
@@ -193,6 +195,21 @@ def _add_symbol_to_star_imports(self, imported_symbol):
...
+            k.split(f"{self._module_name}.")[1]: k
+            for k in self.symbols
+            if k != self._module_name
+        }
+        output_symbols = self.symbols
+        for k, v in output_symbols.items():
+            volume = v["data"].get("symbols_in_volume")
+            if volume:
+                for bad_func_name in volume & set(stripped_names):
+                    volume.remove(bad_func_name)
```
Comment (dacamo76) [inline symbol_exporter/ast_symbol_extractor.py:209]: We might also want to remove symbol from undeclared symbols here.

File: symbol_exporter/ast_symbol_extractor.py [hunk 2]
```diff
@@ -193,6 +195,21 @@ def _add_symbol_to_star_imports(self, imported_symbol):
         default = dict(type=SymbolType.STAR_IMPORT, data=dict(imports=set()))
         self.symbols.setdefault("*", default)["data"]["imports"].add(imported_symbol)
 
+    def post_process_symbols(self):
```
Comment (dacamo76) [inline symbol_exporter/ast_symbol_extractor.py:198]: Why not make symbols a property and let that be the only entry point into the results?

File: symbol_exporter/ast_symbol_extractor.py [hunk 3]
```diff
@@ -16,6 +16,8 @@ class SymbolType(str, Enum):
     CLASS = "class"
     STAR_IMPORT = "star-import"
 
+    BOUNDING_SURFACE_SYMBOL_TYPES = {MODULE, FUNCTION, CLASS}
```
Comment (dacamo76) [inline symbol_exporter/ast_symbol_extractor.py:19]: where is this used?

**Comment by CJ-Wright** [comment]
I think I've addressed the issues, I'm happy to revisit if you'd like, but I also want to get this round of the database started so we can see where things land.

**Pull request merged**